driverkit/Makefile

```diff
@@ -40,36 +60,53 @@ cleanup_s3:
 	S3_DRIVERS_BUCKET=${S3_DRIVERS_BUCKET} S3_DRIVERS_KEY_PREFIX=${S3_DRIVERS_KEY_PREFIX} utils/cleanup_s3 $(addprefix -v ,$(VERSIONS))
 
 # $(1): pseudo-target name
-# $(2): driver version
-# $(3): config file path
+# $(2): config file path
 define gen_build_targets
-validate/$(1): $(3)
-	utils/checkfiles $(3)
+validate/$(1): $(2)
+	utils/checkfiles $(2)
 
-$(1): validate/$(1) $(3)
-	DRIVERKIT=${DRIVERKIT} S3_DRIVERS_BUCKET=${S3_DRIVERS_BUCKET} S3_DRIVERS_KEY_PREFIX=${S3_DRIVERS_KEY_PREFIX} SKIP_EXISTING=${SKIP_EXISTING} utils/build $(3)
+$(1): validate/$(1) $(2)
+	DRIVERKIT=${DRIVERKIT} S3_DRIVERS_BUCKET=${S3_DRIVERS_BUCKET} S3_DRIVERS_KEY_PREFIX=${S3_DRIVERS_KEY_PREFIX} SKIP_EXISTING=${SKIP_EXISTING} utils/build $(2)
 endef
 
 $(foreach CONFIG,$(CONFIGS),\
-	$(eval INNER := $(patsubst config/%,%,$(CONFIG)))\
-	$(eval VERSION := $(patsubst %/,%,$(dir $(INNER))))\
 	$(eval TARGET := $(patsubst config_%,%,$(subst /,_,$(CONFIG))))\
-	$(eval $(call gen_build_targets,$(TARGET),$(VERSION),$(CONFIG)))\
+	$(eval $(call gen_build_targets,$(TARGET),$(CONFIG)))\
 )
 
 # $(1): driver version
 define gen_publish_targets
 split_$(1)_kernelmodules:
+# root is old x86_64
+# TODO: rm once old format support is dropped
 ifneq ("$(wildcard output/$(1)/*.ko)","")
 	@mkdir -p output/$(1)/kernel-module
 	@mv -f output/$(1)/*.ko output/$(1)/kernel-module
 endif
+ifneq ("$(wildcard output/$(1)/x86_64/*.ko)","")
```

Comment:
Basically, we now search in both root folder (old directory tree) and new $arch subfolders.